zdigness/puttdle
# [TASK] Implement Docker, Sequelize-Typescript ORM, Prettier, eslint, documentation

## Description
The goal of this ticket is to:

1. Implement an ORM to the Database. [Sequelize](https://sequelize.org/docs/v6/getting-started/) will be used to do this
2. Standardize documentation for the service
3. Add Dockerization of services to allow easier use of Postgres
4. Add eslint and prettier for code standardization

## Motivation and Context
[Ticket](https://trello.com/c/vvlllUXG/20-standardize-database-interactions)

> [!NOTE]
>
> This is PR 1 of 2. I am breaking this up into more digestible merges. The next PR will focus on unit/integration testing with jest, so CI. On top of that, I want to implement GitHub workflow as actions. This will help standardize how we are writing tests for our features and make sure we have a solid foundation to not break any existing work.

## How Has This Been Tested?
Docker
1. Utilize Docker compose to validate containers spin up and communicate
Sequelize-Typescript ORM
1. Spin up docker containers
2. Send POST request in to POSTMAN to /api/vi/google-login 

General queries
```sql
-- Get all from Users table --
SELECT *
FROM "Users" as "User";

-- Get all from Scores table --
SELECT *
FROM "Scores" as "Score";
```

See attached video for more detail

## Screenshots (if appropriate):
Utilizing Docker, Pgadmin, and Postman

https://github.com/zdigness/puttdle/assets/23369330/f056b9c4-2db6-4b80-8873-21917e32e6b2
